packages/frontend/core/src/bootstrap/global-error-handler.ts

```diff
@@ -0,0 +1,46 @@
+import errorImg from './error.png';
+
+function handleGlobalUnrecoverableError(e: ErrorEvent | PromiseRejectionEvent) {
```

Comment:
I am not sure - some unhandled errors/rejections may not be that disastrous and make the app unusable? The error prompting may be misleading because it may only happend during bootstrap phase some point after app is rendered.

Reply:
errors in renderer should be caught by react error boundary

Reply:
I tried by throwing an error in a service contructor. It seems this is captured in the new handler, instead of in the react error boundary

Reply:
this is bad, error boundary won't stop propagation of errors by default

Reply:
![Simulator 2024-10-16 17.35.33.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/5c0ymolP9B7QStCsS1RP/45663e48-6f9a-41d6-b11a-2cfadc46bced.png)